File: app/src/main/java/br/com/alura/leilao/model/formatter/FormatadorDeMoeda.java
package br.com.alura.leilao.model.formatter;

import java.text.NumberFormat;
import java.util.Locale;

import br.com.alura.leilao.model.Lance;

public class FormatadorDeMoeda{

   private static final Locale BRASIL = new Locale("pt", "BR");

   public static String formata(double valor) {
      NumberFormat formatador = NumberFormat.getCurrencyInstance(BRASIL);
      return formatador.format(valor);
   }
}

File: app/src/main/java/br/com/alura/leilao/model/formatter/FormatadorDeMoeda.kt
package br.com.alura.leilao.model.formatter

import java.text.Normalizer
import java.text.NumberFormat
import java.util.Locale

object FormatadorDeMoeda {

    private val BRASIL = Locale("pt", "BR")

    fun formata(valor: Double): String {
        val valorFormatado = NumberFormat.getCurrencyInstance(BRASIL).format(valor)
        val valorNormalizado = Normalizer.normalize(valorFormatado, Normalizer.Form.NFKC)
        return valorNormalizado
    }
}